### ISSUE TITLE
SuppressWarningsHolder should allow usage of simple name of Checks in aliasList

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#11641]
https://checkstyle.org/config_annotation.html#SuppressWarningsHolder

```
$ cat Test.java
class Test {
  @SuppressWarnings("paramnum")
  public void needsLotsOfParameters( int a, 
  int b, int c, int d, int e, int f, int g, int h) {
    // ...
  }
}

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
       "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
	<property name="charset" value="UTF-8" />
	
   
<module name="TreeWalker">
   <module name="ParameterNumber">
   </module>
   
   <module name="SuppressWarningsHolder">
        <property name="aliasList" value="ParameterNumber=paramnum" />
  </module>
</module>

   <module name="SuppressWarningsFilter" />
</module>

$ java -jar checkstyle-10.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/kevin/Desktop/C_Style/Test.java:4:15: More than 7 parameters (found 8). [ParameterNumber]
Audit done.
```

Expected: no violation.

workaround is to use fully qualified name of Check.
see details - https://github.com/checkstyle/checkstyle/pull/11542#issuecomment-1126821468

**Comment by Kevin222004** [comment]
we have to create a method in https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java
am i right

**Comment by romani** [comment]
Create unit test method to reproduce problem by Input file first. 
Second , update holder to have in map of names simple names also .

**Comment by rnveach** [comment]
https://checkstyle.org/config_annotation.html#SuppressWarningsHolder_Properties
>  The attribute is the fully qualified name of the Check and value is its alias. 

Documentation states that it must be the fully qualified name.

What documentation may not state, is that it looks like it can also be the Check's ID instead of the name.
https://github.com/checkstyle/checkstyle/blob/399f1b48b79731d65e824eb927971ef6174c3463/src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java#L242

I am ok to expand this to simple check names. Making a FQC name seems a bit excessive for simple uses, but they should fall back on that to avoid conflicts. We have a conflicting check between sevntu and checkstyle.

We need to expand documentation with examples of configuration. Current documentation shows no configuration examples to connect with the Java code. It is hard to determine how to configure check even before this change. Another example is what is the default behavior of this check with no aliases set.

@romani Are you look for `CheckNameCheck` or `CheckName` or both?

**Comment by rnveach** [comment]
> Another example is what is the default behavior of this check with no aliases set.

Looks like it does default to the simple check name as seen at:
https://github.com/checkstyle/checkstyle/blob/399f1b48b79731d65e824eb927971ef6174c3463/src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java#L161

**Comment by romani** [comment]
>Are you looking to match against CheckNameCheck or CheckName or both?

This is good point we need both, user need to use same name of Check as he has in his config, so it can be without "Check" suffix.

**Comment by rnveach** [comment]
@strkkk Current behavior says right now if the module name is supplied, it suppresses everything regardless if the IDs are different. See my code past above.

**Comment by 524130120** [comment]
I am on it.

**Comment by 524130120** [comment]
I try to implement that the user can use SuppressWarningsHolder by input both module name(e.g ParameterNumber) and the set alias name(e.g paramnum) without case sensitive. 

The name matching happens at:
https://github.com/checkstyle/checkstyle/blob/51408f83b03311e2c4a86cb28410d4bd7b3388be/src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java#L227-L248
The method `getAlias` at `line 230` wants to return a ailas name or a module name in lowercase. I think it is not enough to use only `checkAlias ` to represent module name and alias name. 

Besides, since the user alway use a module name, we need a method to get a module name of the Checker(e.g get 'ParameterNumber' from 'ParameterNumberCheck')

**Comment by Kevin222004** [comment]
I am on it

**Comment by Kevin222004** [comment]
@romani according to the test https://github.com/checkstyle/checkstyle/blob/74d8802acfb8833cd34538447d0852baa58e6c5b/src/test/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolderTest.java#L154

it is working as we need but https://github.com/checkstyle/checkstyle/issues/11641#issue-1236303060 it is not working 

I am not understanding what changes should i make

**Comment by romani** [comment]
Please make test with values `name="aliasList" value="ParameterNumber=paramnum"` and will fail to suppress violation, so it is still not working.

**Comment by Kevin222004** [comment]
https://github.com/checkstyle/checkstyle/pull/11542#issuecomment-1126821468
This is not getting true in all condition I have tested this with other checkName
```
cat Test.java
class Test {
  
    @SuppressWarnings("const") 
    private static final int a = 0;
}

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
       "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
	<property name="charset" value="UTF-8" />
	
   
<module name="TreeWalker">
   <module name="ConstantName">
   </module>
   
   <module name="SuppressWarningsHolder">
        <property name="aliasList" value="com.puppycrawl.tools.checkstyle.checks.naming.constantname=const"/>
  </module>
</module>

   <module name="SuppressWarningsFilter" />
</module>

kevin@inspiron-15-5510:~/Desktop/check_style$ java -jar checkstyle-10.3.3-all.jar  -c config.xml Test.java 
Starting audit...
[ERROR] /home/kevin/Desktop/check_style/Test.java:4:30: Name 'a' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
Audit done.
Checkstyle ends with 1 errors.

```

**Comment by Kevin222004** [comment]
if you agree I have made few changes according to them we can use this property like this 
```
$ cat Test.java

class Test {
  
    @SuppressWarnings("constantname=const") 
    private static final int a = 0;
}

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
       "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
	<property name="charset" value="UTF-8" />
	
   
<module name="TreeWalker">
   <module name="ConstantName">
   </module>
   
   <module name="SuppressWarningsHolder">
        <property name="aliasList" value="constantname=const"/>
  </module>
</module>

   <module name="SuppressWarningsFilter" />
</module>
kevin@inspiron-15-5510:~/Desktop/check_style$ java -jar /home/kevin/Desktop/check_style/checkstyle/target/checkstyle-10.3.4-SNAPSHOT-all.jar  -c config.xml Test.java 
Starting audit...
Audit done.
```

**Comment by Kevin222004** [comment]
```
$ cat Test.java
class Test {
  @SuppressWarnings("paramnum")
  public void needsLotsOfParameters( int a, 
  int b, int c, int d, int e, int f, int g, int h) {
    // ...
  }
}

$cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
       "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
	<property name="charset" value="UTF-8" />
	
   
<module name="TreeWalker">
   <module name="ParameterNumber">
   </module>
   
   <module name="SuppressWarningsHolder">
        <property name="aliasList" value="ParameterNumber=paramnum" />
  </module>
</module>

   <module name="SuppressWarningsFilter" />
</module>
kevin@inspiron-15-5510:~/Desktop/check_style$ java -jar /home/kevin/Desktop/check_style/checkstyle/target/checkstyle-10.3.4-SNAPSHOT-all.jar  -c config.xml Test.java 
Starting audit...
Audit done.
```

**Comment by romani** [comment]
Yes, this what is expected.
Please send PR.

**Comment by Kevin222004** [comment]
@romani 

> Yes, this what is expected. Please send PR.

please give a look to https://github.com/checkstyle/checkstyle/issues/11641#issuecomment-1249681366 once again I have achieved this not the expectation like https://github.com/checkstyle/checkstyle/issues/11641#issue-1236303060 

I am not understanding how user will provide the alias to us
 ex:- for
```  <property name="aliasList" value="ParameterNumber=paramnum" />```
in the original project code how user will define SuppressionWarningHolder as paramnum

**Comment by Harsh-Agrawal96** [comment]
Is anything left in this issue, I think the fix is merged.

**Comment by romani** [comment]
Fix is merged
